server/runtime/src/main/java/org/infinispan/server/tasks/DistributedServerTask.java

```diff
@@ -30,14 +26,14 @@
    @ProtoField(2)
    final String cacheName;
 
-   @ProtoField(number = 3, collectionImplementation = ArrayList.class)
-   final List<TaskParameter> parameters;
+   @ProtoField(3)
```

Comment:
The correct way to handle this is to use a new field. Ideally we would just be able to add `@ProtoReservice(value=3)` to the class and `proto.lock` checks would pass. However, the `reserved` statement is currently commented due to a limitation in the protostream proto parser https://github.com/infinispan/protostream/issues/174.

Adding `@ProtoReservice(value=3)` will still prevent the field being used in the `ProtoField` though.

A workaround to keep proto compatibility would be todo the following:

```java
   @ProtoField(1)
   final String taskName;

   @ProtoField(2)
   final String cacheName;

   @ProtoField(number = 3, collectionImplementation = ArrayList.class)
   List<TaskParameter> parameters;

   @ProtoField(4)
   final TaskContext context;

   @ProtoFactory
   DistributedServerTask(String taskName, String cacheName, List<TaskParameter> parameters, TaskContext context) {
      this(taskName, cacheName, context);
   }
   
   public DistributedServerTask(String taskName, String cacheName, TaskContext context) {
      this.cacheName = cacheName;
      this.taskName = taskName;
      this.context = context;
   }
```

Reply:
This is for internal invocation only, so I'm not fussed about losing backwards compat.